docs/errors/usage_errors.md

```diff
@@ -1238,3 +1238,105 @@ except PydanticUserError as exc_info:
 
 [related specification section]: https://typing.readthedocs.io/en/latest/spec/callables.html#unpack-for-keyword-arguments
 [PEP 692]: https://peps.python.org/pep-0692/
+
+
+
+## Unsupported type for `validate_call` {#validate-call-type}
+
+`validate_call` has some limitations on the types it can validate. This error is raised when you try to use it with an unsupported type. Currently the supported types are:
+`LambdaType, FunctionType, MethodType, functools.partial`. For the case of `functools.partial`, the function being partially applied must be one of the supported types.
+
+
+### `@classmethod`, `@staticmethod`, and `@property`
+
+These decorators cannot be put before `validate_call` because they return a class rather than one of the supported types. The correct way to use them is to put `validate_call` before the decorator.
+
+```py
+from pydantic import PydanticUserError, validate_call
+
+# error
+try:
+
+    class A:
+        @validate_call
+        @classmethod
+        def f1(cls): ...
+
+except PydanticUserError as exc_info:
+    assert exc_info.code == 'validate-call-type'
+
+
+# correct
+@classmethod
+@validate_call
+def f2(cls): ...
+```
+
+
+### Classes
+
+To avoid ambiguity, `validate_call` should be applied to methods, not classes. If you want to validate the constructor of a class, you should put `validate_call` on top of `__init__` or `__new__` instead.
+
+```py
+from pydantic import PydanticUserError, validate_call
+
+# error
+try:
+
+    @validate_call
+    class A1: ...
+
+except PydanticUserError as exc_info:
+    assert exc_info.code == 'validate-call-type'
+
+
+# correct
+class A2:
+    @validate_call
+    def __init__(self): ...
+
+    @validate_call
+    def __new__(cls): ...
+```
+
+### Custom callable
+
+Although you can create custom callable types in Python with `__call__`, currently the instance of these types cannot be validated with `validate_call`. This may change in the future, but for now, you should use `validate_call` explicitly on `__call__` instead.
```

Comment:
```suggestion
Although you can create custom callable types in Python by implementing a `__call__` method, currently the instances of these types cannot be validated with `validate_call`. This may change in the future, but for now, you should use `validate_call` explicitly on `__call__` instead.
```